test/Condenser.Tests.Integration/LeaderElectionFacts.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Condenser.Tests.Integration
+{
+    public class LeaderElectionFacts
+    {
+        [Fact]
+        public async void TestLeaderLeaving()
+        {
+            var server1 = new MiniServer(8888);
+            var server2 = new MiniServer(9999);
+
+            var regClient1 = new CondenserDotNet.Client.ServiceRegistrationClient();
+            regClient1.Config(serviceName: "TestService1", serviceId: "ServiceId1", address: "localhost", port: 8888);
+            regClient1.AddUrls("api/testurl");
+            regClient1.AddHealthCheck("/health",2,2);
+            regClient1.AddLeaderElectionKey("/electionKey/Test");
+            await regClient1.RegisterServiceAsync();
+
+            //We should be the leader lets wait on that
+            await regClient1.Leader;
+
+            //Next we setup the second service
+
+            var regClient2 = new CondenserDotNet.Client.ServiceRegistrationClient();
+            regClient2.Config(serviceName: "TestService1", serviceId: "ServiceId2", address: "localhost", port: 9999);
+            regClient2.AddUrls("api/testurl");
+            regClient2.AddLeaderElectionKey("/electionKey/Test");
+            regClient2.AddHealthCheck("/health", 2, 2);
+            await regClient2.RegisterServiceAsync();
+
+            //Because this is a closure it will be false if the awaiter finishes before the leader election does
+            bool shouldBeLeader = false;
+            var wait = new ManualResetEvent(false);
+
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+            Task.Run(async () =>
+            {
+                await regClient2.Leader;
+                Assert.True(shouldBeLeader);
+                wait.Set();
+             });
+#pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
```

Comment:
This never gets hit, however watching consul, the second leader is correctly taking over after the shutdown of the health check on the first service